### PR TITLE
HPCC-19369 Pyrolite serialization and de-serialization

### PR DESCRIPTION
**Pull request by jpmcmu**
- Register / use builtin Python picklers with GenericRowWithSchema
- Implemented a getRDD version that allows PySpark to create RDDs

Signed-off-by: James McMullan <James.McMullan@lexisnexis.com>

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java [hunk 1]
```diff
@@ -42,6 +44,11 @@
   private RecordDef recordDefinition;
   private boolean isIndex;
 
+  // Make sure Python picklers have been registered
+  static {
+      EvaluatePython.registerPicklers();
```
Comment (rpastrana) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java]: Can this fail?

Comment (jpmcmu) [reply DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java]: Good question. That function does not throw any exceptions or return any error codes. If it were to fail due to a bug in Spark the user would get an error about being unable to serialize objects.

Comment (jpmcmu) [reply DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java]: @rpastrana The PySpark job fails if the registerPicklers method fails

**Pull request merged**